sdk/core/Azure.Core/src/Pipeline/BearerTokenAuthenticationPolicy.cs

```diff
@@ -278,93 +241,103 @@ public async ValueTask<string> GetHeaderValueAsync(HttpMessage message, TokenReq
                         }
 
                         // We were waiting on a previous headerValueTcs operation which was canceled.
-                        //Retry the call to GetTaskCompletionSources.
+                        //Retry the call to RefreshTokenRequestState.
                         continue;
                     }
                 }
             }
 
-            private (TaskCompletionSource<HeaderValueInfo> InfoTcs, TaskCompletionSource<HeaderValueInfo>? BackgroundUpdateTcs, bool GetTokenFromCredential)
-                GetTaskCompletionSources(TokenRequestContext context)
+            /// <summary>
+            /// Refresh the <see cref="TokenRequestState"/> for the current request context.
+            /// </summary>
+            /// <param name="context">The <see cref="TokenRequestContext"/> for the current request.</param>
+            /// <param name="updatedState"></param>
+            /// <returns>true if the token should be refreshed from the credential, else false.</returns>
+            private bool RefreshTokenRequestState(TokenRequestContext context, out TokenRequestState updatedState)
             {
                 // Check if the current state requires no updates to _state under lock and is valid.
                 // All checks must be done on the local prefixed variables as _state can be modified by other threads.
                 var localState = _state;
```

Comment:
nit: per desire for clarity, would it make sense to use type names instead of `var` here?  var seems ok in an assignment when the type name is on the right, but without intellisense, it's harder to understand in e.g. a PR review.

Reply:
Also curious on this one (but non-blocking and non-urgent) -- why is localState needed separately from _state or updatedState?

Reply:
VAR ALL THE THINGS!